python/lsst/sims/catUtils/mixins/sncat.py

```diff
@@ -0,0 +1,302 @@
+        '''
+        # self._suppressDimSN = True
+        if not hasattr(self, '_suppressDimSN'):
+            suppressDimSN_default = True
+            self._suppressDimSN = suppressDimSN_default
+        return self._suppressDimSN
+
+    @suppressDimSN.setter
+    def suppressDimSN(self, suppressDimSN):
+        '''
+        set the value of suppressDimSN of the catalog
+
+        Parameters
+        ----------
+        value : Boolean, mandatory
+            Value to set suppressDimSN to
+        '''
+        # if suppressDimSN is None:
+        #    self._suppressDimSN = True
+        # else:
+        self._suppressDimSN = suppressDimSN
+        return self._suppressDimSN
+
+    @property
+    def suppressHighzSN(self):
+        '''
+        Boolean to decide whether to output information corresponding to SN
+        at redshift above self.maxz
+        '''
+
+        return True
+
+    @property
+    def maxTimeSNVisible(self):
+        '''
+        The catalog will provide values for SN flux (even if zero according to
+        model) for peak \pm maxTimeVisibilityforSN
+        '''
+        return 100.
+
+    @property
+    def maxz(self):
+        return 1.2
+
+    @astropy.utils.lazyproperty
+    def photometricparameters(self, expTime=15., nexp=2):
+        lsstPhotometricParameters = PhotometricParameters(exptime=expTime,
+                                                          nexp=nexp)
+        return lsstPhotometricParameters
+
+    @astropy.utils.lazyproperty
+    def lsstBandpassDict(self):
+        return BandpassDict.loadTotalBandpassesFromFiles()
+
+    @astropy.utils.lazyproperty
+    def observedIndices(self):
+        bandPassNames = self.obs_metadata.bandpass
+        return [self.lsstBandpassDict.keys().index(x) for x in bandPassNames]
+
+    @astropy.utils.lazyproperty
+    def lsstpbase(self):
+        pbase = PhotometryBase()
+        return pbase
+
+    def get_snid(self):
+        # Not necessarily unique if the same galaxy hosts two SN
+        # Use refIdCol to access the relevant id column of the dbobj
+        # Should revert to galTileID for galaxyTiled catalogDBObj and
+        # id for galaxyObj catalogDBObj
+        # (email from Scott)
+        return self.column_by_name(self.refIdCol)
+
+    @property
+    def numobjs(self):
+        return len(self.column_by_name('id'))
+
+    @compound('snra', 'sndec', 'z', 'vra', 'vdec', 'vr')
+    def get_angularCoordinates(self):
+        '''
+        Obtain the coordinates and velocity of the SN from the host galaxy
+
+        Returns
+        -------
+        `np.ndarray` of coordinara, dec, z, vra, vdec, and vr
+
+        '''
+        hostra, hostdec, hostz = self.column_by_name('raJ2000'),\
+            self.column_by_name('decJ2000'),\
+            self.column_by_name('redshift')
+        snra, sndec, snz, snvra, snvdec, snvr = self.SNCoordinatesFromHost(
+            hostra, hostdec, hostz)
+
+        return ([snra, sndec, snz, snvra, sndec, snvr])
```

Comment:
sndec appears twice in the reutrn np.array.  Second time is supposed to be snvdec.